ee/tuf/autoupdate.go

```diff
@@ -520,15 +568,57 @@ func (ta *TufAutoupdater) downloadUpdate(binary autoupdatableBinary, targets dat
 
 		// The release is already available in the library and it's not our current running version --
 		// return the version to signal for a restart.
-		return release, nil
+		return target, nil
 	}
 
 	// We haven't yet downloaded this release -- download it
-	if err := ta.libraryManager.AddToLibrary(binary, currentVersion, release, releaseMetadata); err != nil {
-		return "", fmt.Errorf("could not add release %s for binary %s to library: %w", release, binary, err)
+	if err := ta.libraryManager.AddToLibrary(binary, currentVersion, target, targetMetadata); err != nil {
+		return "", fmt.Errorf("could not add target %s for binary %s to library: %w", target, binary, err)
+	}
+
+	return target, nil
+}
+
+// findTarget selects the appropriate target from `targets` for the given binary, using the pinned version (if set)
+// and otherwise selecting the correct release for the given channel.
+func findTarget(ctx context.Context, binary autoupdatableBinary, targets data.TargetFiles, pinnedVersion string, channel string, slogger *slog.Logger) (string, data.TargetFileMeta, error) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	if pinnedVersion != "" {
+		target, targetMetadata, err := findTargetByVersion(context.Background(), binary, targets, pinnedVersion)
```

Comment:
just curious for myself why the use of `context.Background()` over the parent `ctx` here?

Reply:
Because I copy-pasted from somewhere and neglected to change it 😅 

Updated, thanks!